### PR TITLE
Pod Events

### PR DESCRIPTION
**Pull request by adfost**
Screen Shot 2021-06-11 at 5 22 15 PM

### CODE REVIEW

File: pkg/acl/endpoints.go [hunk 1]
```diff
@@ -44,10 +44,6 @@ var (
 	tenants                     = "/tenants"
 	tenantsDetail               = "/namespaces/:tenantNamespace/tenants/:tenantName"
 	podsDetail                  = "/namespaces/:tenantNamespace/tenants/:tenantName/pods/:podName"
-	tenantsDetailSummary        = "/namespaces/:tenantNamespace/tenants/:tenantName/summary"
-	tenantsDetailPods           = "/namespaces/:tenantNamespace/tenants/:tenantName/pods"
-	tenantsDetailPools          = "/namespaces/:tenantNamespace/tenants/:tenantName/pools"
-	tenantsDetailLicense        = "/namespaces/:tenantNamespace/tenants/:tenantName/license"
```
Comment (bexsoft) [inline pkg/acl/endpoints.go:50]: These are required for tenant details, please do not remove

File: pkg/acl/endpoints.go [hunk 2]
```diff
@@ -322,17 +318,13 @@ var endpointRules = map[string]ConfigurationActionSet{
...
-	tenantsDetail:        tenantsActionSet,
-	tenantsDetailSummary: tenantsActionSet,
-	tenantsDetailPods:    tenantsActionSet,
-	tenantsDetailPools:   tenantsActionSet,
-	tenantsDetailLicense: tenantsActionSet,
-	podsDetail:           tenantsActionSet,
-	storage:              storageActionSet,
-	storageDrives:        storageActionSet,
-	storageVolumes:       storageActionSet,
-	license:              licenseActionSet,
```
Comment (bexsoft) [inline pkg/acl/endpoints.go:335]: These are required for tenant details, please do not remove

File: portal-ui/src/screens/Console/Console.tsx [hunk 3]
```diff
@@ -358,22 +358,6 @@ const Console = ({
...
-      path: "/namespaces/:tenantNamespace/tenants/:tenantName/pods",
-    },
-    {
-      component: TenantDetails,
-      path: "/namespaces/:tenantNamespace/tenants/:tenantName/pools",
-    },
-    {
-      component: TenantDetails,
-      path: "/namespaces/:tenantNamespace/tenants/:tenantName/license",
-    },
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Console.tsx:376]: These are required for tenant details, please do not remove

File: portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx [hunk 4]
```diff
@@ -201,19 +245,17 @@ const TenantDetails = ({
...
-        `/api/v1/namespaces/${tenantNamespace}/tenants/${tenantName}/pods/${podName}`
-      )
-      .then((res: string) => {
-        setLogLines(res.split("\n"));
-      })
-      .catch((err) => {
-        setErrorSnackMessage(err);
-      });
-  }, [tenantNamespace, tenantName, podName, setErrorSnackMessage]);
+    loadInfo();
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx:256]: Please use the content of loadInfo function inside this effect and remove the `// eslint-disable-next-line react-hooks/exhaustive-deps`. I think this change shouldn't go

File: portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx [hunk 5]
```diff
@@ -201,19 +245,17 @@ const TenantDetails = ({
...
-        "GET",
-        `/api/v1/namespaces/${tenantNamespace}/tenants/${tenantName}/pods/${podName}`
-      )
-      .then((res: string) => {
-        setLogLines(res.split("\n"));
-      })
-      .catch((err) => {
-        setErrorSnackMessage(err);
-      });
-  }, [tenantNamespace, tenantName, podName, setErrorSnackMessage]);
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx:216]: Please use the content of loadInfo function inside this effect and remove the `// eslint-disable-next-line react-hooks/exhaustive-deps`. I think this change shouldn't go

File: portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx [hunk 6]
```diff
@@ -225,44 +267,82 @@ const TenantDetails = ({
             </Link>
             {" > "}
             <Link
-              to={`/namespaces/${tenantNamespace}/tenants/${tenantName}`}
+              to={`/namespaces/${match.params["tenantNamespace"]}/tenants/${match.params["tenantName"]}`}
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx:270]: Please use tenantNamespace & tenantName variables

File: portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx [hunk 7]
```diff
@@ -225,44 +267,82 @@ const TenantDetails = ({
...
               className={classes.breadcrumLink}
             >
-              {tenantName}
+              {match.params["tenantName"]}
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx:273]: please use tenantName var

File: portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx [hunk 8]
```diff
@@ -225,44 +267,82 @@ const TenantDetails = ({
...
             </Link>
-            {` > Pods > ${podName}`}
+            {` > Pods > ${match.params["podName"]}`}
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx:275]: Please use podName var

File: restapi/admin_tenants.go [hunk 9]
```diff
@@ -1408,7 +1416,7 @@ func getTenantPodsResponse(session *models.Principal, params admin_api.GetTenant
 	}
 	retval := []*models.TenantPod{}
 	for _, pod := range pods.Items {
-		var restarts int64
+		var restarts int64 = 0
```
Comment (bexsoft) [inline restapi/admin_tenants.go:1419]: Please check gofmt checks

**Review by bexsoft — CHANGES_REQUESTED** [review]

File: pkg/acl/endpoints.go (hunk 1 above)
Comment (dvaldivia) [reply pkg/acl/endpoints.go:50]: I thought we fix this together @adfost , once you fix this, rebase  and squash to avoid this problem

**Comment by dvaldivia** [comment]
Can you attach a screenshot of how it looks?

**Comment by dvaldivia** [comment]
Can you sort the events in ascending order (older on top, newer at the bottom) and group them?

**Review by dvaldivia — CHANGES_REQUESTED** [review]
Please fix all the warnings introduced on `PodDetails.tsx`

```
src/screens/Console/Tenants/TenantDetails/PodDetails.tsx
  Line 19:8:    'get' is defined but never used                      @typescript-eslint/no-unused-vars
  Line 31:3:    'Button' is defined but never used                   @typescript-eslint/no-unused-vars
  Line 32:3:    'IconButton' is defined but never used               @typescript-eslint/no-unused-vars
  Line 33:3:    'Menu' is defined but never used                     @typescript-eslint/no-unused-vars
  Line 34:3:    'MenuItem' is defined but never used                 @typescript-eslint/no-unused-vars
  Line 48:8:    'history' is defined but never used                  @typescript-eslint/no-unused-vars
  Line 49:10:   'LogMessage' is defined but never used               @typescript-eslint/no-unused-vars
  Line 50:10:   'IPolicyItem' is defined but never used              @typescript-eslint/no-unused-vars
  Line 51:10:   'TabPanel' is defined but never used                 @typescript-eslint/no-unused-vars
  Line 167:10:  'selectedTab' is assigned a value but never used     @typescript-eslint/no-unused-vars
  Line 167:23:  'setSelectedTab' is assigned a value but never used  @typescript-eslint/no-unused-vars
  Line 168:19:  'setLoading' is assigned a value but never used      @typescript-eslint/no-unused-vars
  Line 169:10:  'log' is assigned a value but never used             @typescript-eslint/no-unused-vars

```

**Comment by dvaldivia** [comment]
While doing  `kubectl get event` I'm seeing the last event, `4 minutes ago` but the UI is showing `6 hours ago`
```
➜ k get event -n ns-2 --field-selector involvedObject.name=asdzxczxc-pool-0-3
LAST SEEN   TYPE      REASON             OBJECT                   MESSAGE
4m41s       Warning   FailedScheduling   pod/asdzxczxc-pool-0-3   0/5 nodes are available: 1 node(s) had taint {node-role.kubernetes.io/master: }, that the pod didn't tolerate, 1 node(s) were unschedulable, 3 node(s) had volume node affinity conflict.
```

Screen Shot 2021-06-14 at 10 12 24 AM


The problem is also present in pod listing, where the age of pods is not aligned with theUI

```
➜ k get pods -n ns-1
NAME                                     READY   STATUS    RESTARTS   AGE
asdasd-console-5c5f969756-ghd98          1/1     Running   0          12d
asdasd-log-0                             1/1     Running   0          12d
asdasd-log-search-api-657dffbcb8-9ln74   1/1     Running   3          12d
asdasd-pool-0-0                          1/1     Running   0          7m3s
asdasd-pool-0-1                          1/1     Running   0          12d
asdasd-pool-0-2                          1/1     Running   0          12d
asdasd-pool-0-3                          1/1     Running   0          12d
asdasd-prometheus-0                      2/2     Running   0          12d
```

Screen Shot 2021-06-14 at 10 17 35 AM

File: swagger.yml [hunk 10]
```diff
@@ -5219,4 +5248,23 @@ definitions:
...
+    type: array
+    items:
+      $ref: "#/definitions/eventListElement"
 
+  eventListElement:
+    type: object
+    properties:
+      namespace:
+        type: string
+      lastseen:
```
Comment (dvaldivia) [inline swagger.yml:5261]: rename this to `last_seen`
Change type to `integer` of `format: int64`

File: restapi/admin_tenants.go [hunk 11]
```diff
@@ -1439,6 +1446,31 @@ func getPodLogsResponse(session *models.Principal, params admin_api.GetPodLogsPa
...
+	/*obj, err := clientset.CoreV1().Pods(params.Namespace).Get(ctx, params.PodName, metav1.GetOptions{})
+	if err != nil {
+		return nil, prepareError(err)
+	}*/
+	events, _ := clientset.CoreV1().Events(params.Namespace).List(ctx, metav1.ListOptions{})
+	retval := models.EventListWrapper{}
+	for i := 0; i < len(events.Items); i++ {
+		retval = append(retval, &models.EventListElement{
+			Namespace: events.Items[i].Namespace,
+			Lastseen:  fmt.Sprintf("%d", events.Items[i].LastTimestamp.Unix()),
```
Comment (dvaldivia) [inline restapi/admin_tenants.go:1465]: the `.Unix()` already returns `int64`, store the integer that way

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: portal-ui/src/screens/Console/Account/Account.tsx [hunk 12]
```diff
@@ -101,9 +101,8 @@ const Account = ({
   const [filter, setFilter] = useState<string>("");
   const [addScreenOpen, setAddScreenOpen] = useState<boolean>(false);
   const [deleteOpen, setDeleteOpen] = useState<boolean>(false);
-  const [selectedServiceAccount, setSelectedServiceAccount] = useState<
-    string | null
-  >(null);
+  const [selectedServiceAccount, setSelectedServiceAccount] =
```
Comment (dvaldivia) [inline portal-ui/src/screens/Console/Account/Account.tsx:104]: this needs to be formatted with prettier `2.3.1`

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: yarn.lock [hunk 13]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (bexsoft) [inline yarn.lock:1]: Please remove this file

File: portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx (hunk 4 above)
Comment (bexsoft) [reply portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx:252]: Please review this

**Review by bexsoft — CHANGES_REQUESTED** [review]

File: portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx [hunk 14]
```diff
@@ -148,12 +154,48 @@ const TenantDetails = ({
...
   const podName = match.params["podName"];
 
+  const loadInfo = () => {
+    api
+      .invoke(
+        "GET",
+        `/api/v1/namespaces/${tenantNamespace}/tenants/${tenantName}/pods/${podName}`
+      )
+      .then((res: string) => {
+        setLog(res);
```
Comment (dvaldivia) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx]: remove `setLog` since `log` is not used

**Comment by dvaldivia** [comment]
Screen Shot 2021-06-15 at 10 46 47 AM

I'm getting events that are not related to the pod in question

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: consoleAdmin.json [hunk 15]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (dvaldivia) [inline consoleAdmin.json:1]: please remove this file

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx [hunk 16]
```diff
@@ -148,6 +153,10 @@ const TenantDetails = ({
   match,
   setErrorSnackMessage,
 }: ITenantDetailsProps) => {
+  const [loading, setLoading] = useState<boolean>(false);
+  const [log, setLog] = useState<string>("");
```
Comment (dvaldivia) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx:157]: remove unused lines

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx [hunk 17]
```diff
@@ -234,33 +264,71 @@ const TenantDetails = ({
...
+            <TableWrapper
+              itemActions={[]}
+              columns={[
+                { label: "Namespace", elementKey: "namespace" },
+                { label: "Last Seen", elementKey: "seen" },
+                { label: "Message", elementKey: "message" },
+                { label: "Event Type", elementKey: "event_type" },
+                { label: "Reason", elementKey: "reason" },
+              ]}
+              isLoading={false}
```
Comment (dvaldivia) [inline portal-ui/src/screens/Console/Tenants/TenantDetails/PodDetails.tsx:325]: do show a loading indicator

**Review by dvaldivia — CHANGES_REQUESTED** [review]

**Review by dvaldivia — DISMISSED** [review]

**Review by dvaldivia — CHANGES_REQUESTED** [review]
don't delete md files

**Review by dvaldivia — DISMISSED** [review]

**Review by dvaldivia — APPROVED** [review]

**Review by bexsoft — APPROVED** [review]

**Pull request merged**